File: Python/bll/book_category_manager.py
from dal.book_category_service import BookCategoryService
from models.book_category import BookCategory


class BookCategoryManager:
    def __init__(self):
        self.category_service = BookCategoryService()

    def get_all_categories(self):
        return self.category_service.get_all_categories()

    def add_category(self, category: BookCategory) -> bool:
        if not category.category_name:
            raise Exception("类别名称不能为空！")
        return self.category_service.add_category(category) > 0

    def update_category(self, category: BookCategory) -> bool:
        if not category.category_name:
            raise Exception("类别名称不能为空！")
        return self.category_service.update_category(category) > 0

    def delete_category(self, category_id: int) -> bool:
        return self.category_service.delete_category(category_id) > 0

File: BM.BLL/BookCategoryManager.cs
using BookManagement.Model;
using BookManagement.DAL;

namespace BookManagement.BLL
{
    public class BookCategoryManager
    {
        private BookCategoryService categoryService = new BookCategoryService();

        public System.Data.DataTable GetAllCategories()
        {
            return categoryService.GetAllCategories();
        }

        public bool AddCategory(BookCategory category)
        {
            // 业务规则验证
            if (string.IsNullOrEmpty(category.CategoryName))
            {
                throw new System.Exception("类别名称不能为空！");
            }
            return categoryService.AddCategory(category) > 0;
        }

        public bool UpdateCategory(BookCategory category)
        {
            if (string.IsNullOrEmpty(category.CategoryName))
            {
                throw new System.Exception("类别名称不能为空！");
            }
            return categoryService.UpdateCategory(category) > 0;
        }

        public bool DeleteCategory(int categoryId)
        {
            return categoryService.DeleteCategory(categoryId) > 0;
        }
    }
}
